pages/your_tasks/real_dev_environments.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Fast, Declarative, Reproducible and Composable Developer Environments using Nix 
+description: "Going a step beyond just \"virtual envirnments\" - with advantages and drawbacks"
+contributors:
+  - "Nicole Dresselhaus"
+page_id: developer_environments
+related_pages:
+  your_tasks: 
+    - virtual_environments
+---
+
+## How are devenvs different from virtual software development environments?
+
+Developer Environments (short {% tool "devenv" %}s) are a more holistic
+approach to the classical virtual environments. Virtual environments - despite
+their undisputed advantages - tend to break at similar steps in the real world.
+An environment working on one machine might break on another because of
+different operating system, the existence or absence of globally installed
+packages, different hardware etc.
+You local software might behave different as a docker-container, yield different
+results or may not run at all.
+
+## Advantages and Drawbacks
+
+### Advantages
+
+- The same definition runs nearly everywhere the same.
+- Generating a docker-image of your software is included
+- Automatically setting up multiple services that get started (and even seeded)
+  automatically and can be used reproducibly, i.e.
+  - Setting up a Database-Service in the backgrund
+  - seeding initial data
+  - sharing access via environment-variables
+- Easy to use `git clone ... && cd ... && devenv shell` and you are done (after
+  proper setup)
+
+### Drawbacks
+
+- Can be quite complicated to set up if you stray away from the common schemes
+- Not suitable for "just trying stuff out"
+- As it depends on nix you are forced to learn the nix-language and the
```

Comment:
The particular solution that is propagated here (devenv.sh) is dependent on nix, but is the whole concept of development environments? I cannot imagine that some competitors won't arise. Maybe rephrase a bit this sentence to make sure this is clear.